Audio (#17)
hostmobility/documentation: docs/interfaces/audio.md

---
title: Audio
tags:
  - Audio
  - T30
  - T30 FR
  - CT
  - CT GL
  - C61
  - MX-V
---
## Overview

> Many of our products has audio and microphone for one or both linux system and from the modem. If modem is supported check [modem](modem.md) pages under specific machine.
> On the linux side we use standard [ALSA](https://www.alsa-project.org).

Reviewer: ```suggestion
Many of our platforms feature audio output and microphone input at the operating system level and, in some cases, via the on-board modem. To see if modem audio is supported, please consult the platform-specific [modem](modem.md) pages.

On the Linux side we use standard [ALSA](https://www.alsa-project.org).
```

### T30 FR and T30
This product comes with one 3.5 stereo phone jack with analog stereo audio out and one 3.5 stereo phone jack with mono in (microphone) interface.
See technical specification for more information regarding the connectors.

### CT and CT GL
This product comes with a JAE IL-AG5-14P-D3L2 with analog stereo audio out and with mono in (microphone) interface.
See technical specification for more information regarding the connectors.

### MX-V

> This product comes with a 2x34PIN MINI50 2-BAY 90DEG PIN HEADER with analog stereo audio out and with mono in (microphone) interface. And possibility use HDMI audio from the DVI connector. See technical specification for more information regarding the connectors.

Reviewer: ```suggestion
This product comes with a 2x34PIN MINI50 2-BAY 90DEG PIN HEADER with analog stereo audio out and a mono in (microphone) interface. It's also possible to use HDMI audio from the DVI connector. See the technical specification for more information regarding the connectors.
```

### C61

> This product comes with one 2.5 stereo phone jack with analog stereo audio out and with mono in (microphone) interface. See technical specification for more information regarding the connectors.

Reviewer: ```suggestion
This product comes with one 2.5 stereo phone jack with analog stereo audio out and with mono in (microphone) interface. See the technical specification for more information regarding the connectors.
```

> **Note:** Current status of this function is untested by development apartment but will soon be tested.

Reviewer: ```suggestion
**Note:** This feature is currently untested for modern BSPs.
```

## Examples

> * Example to create a loop back (activate mic and hear yourself)

Reviewer: ```suggestion
* Example on how to create a loop back (activate mic and hear yourself)
```

```bash
arecord -d 5 -D plughw:1 -f dat output.wav & sleep 1 && aplay -D plughw:1 output.wav & echo recording and audio output
```

* Test audio with a sinus tone at 997Hz.
```bash
speaker-test -D plughw:1 -t sine -r 997 -c 1 -l 2

speaker-test -D plughw:1,0 -t sine -r 997 -c 1 -l 2
```

> * list devices with `aplay -L` or `aplay --list-devices` example MX-V

Reviewer: ```suggestion
* list devices with `aplay -L` or `aplay --list-devices` (example from MX-V)
```

```bash
card 0: imxhdmisoc [imx-hdmi-soc], device 0: i.MX HDMI Audio Tx i2s-hifi-0 []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Audiotlv320aic3 [Audio-tlv320aic310x], device 0: 2028000.ssi-tlv320aic3x-hifi tlv320aic3x-hifi-0 []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

```

## For Linux 3.1 system

> **Note:** for bsp release 1.6(1.5) and older

Reviewer: ```suggestion
**Note:** for BSP release 1.6/1.5 and older
```

To direct line-in to line-out:

```bash
amixer -c 2 set 'Left HP Mixer Line Bypass' on
amixer -c 2 set 'Right HP Mixer Line Bypass' on
```

> Playback-hdmi:

Reviewer: ```suggestion
Playback HDMI:
```
